dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/scheduling/DefaultSchedulingManager.java

```diff
@@ -78,21 +83,31 @@
 
     @Autowired
     private MessageService messageService;
-
+    
+    @Autowired
+    private LeaderManager leaderManager;
+    
     private TaskScheduler jobScheduler;
 
     public void setTaskScheduler( TaskScheduler JobScheduler )
     {
         this.jobScheduler = JobScheduler;
     }
-
+    
     private AsyncListenableTaskExecutor jobExecutor;
 
     public void setTaskExecutor( AsyncListenableTaskExecutor jobExecutor )
     {
         this.jobExecutor = jobExecutor;
     }
 
+    @PostConstruct
+    public void init()
+    {
+        leaderManager.setSchedulingManager( this );
```

Comment:
Doesn't it work autowiring SchedulingManager in the LeaderManager?

Reply:
There is a circular dependency there. SchedulingManager already has LeaderManager autowired - to check if the current instance is a leader for a leaderOnlyJob. LeaderManager also needs SchedulingManager to schedule "Leader renewal adhoc jobs". There are a couple of ways to overcome this, one of them is what I implemented. Another way would be do add @Lazy initialization. Or maybe we would have to refactor it completely which I felt was unnecessary.

Reply:
Ok - good.